lib/osf-components/addon/components/editable-field/provider-metadata-manager/component.ts

```diff
@@ -0,0 +1,95 @@
+import { action } from '@ember/object';
+import { alias, and } from '@ember/object/computed';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+import DS from 'ember-data';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+
+import ModeratorModel from 'ember-osf-web/models/moderator';
+import Registration, { ProviderMetadata } from 'ember-osf-web/models/registration';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+
+export interface Args {
+    registration: Registration;
+}
+
+export interface ProviderMetadataManager {
+    save: () => void;
+    cancel: () => void;
+    inEditMode: boolean;
+    currentProviderMetadata: ProviderMetadata[];
+    userCanEdit: boolean;
+}
+
+export default class ProviderMetadataManagerComponent extends Component<Args> {
+    @service currentUser!: CurrentUserService;
+    @service intl!: Intl;
+    @service store!: DS.Store;
+    @service toast!: Toast;
+
+    @task({ withTestWaiter: true })
+    save = task(function *(this: ProviderMetadataManagerComponent) {
+        if (this.args.registration) {
+            try {
+                this.providerSpecificMetadata = this.deepCopy(this.currentProviderMetadata);
+                yield this.args.registration.save();
+            } catch (e) {
+                const errorMessage = this.intl.t('registries.registration_metadata.edit_provider_metadata.error');
+                captureException(e, { errorMessage });
+                this.args.registration.rollbackAttributes();
+                this.toast.error(getApiErrorMessage(e), errorMessage);
+                throw e;
+            }
+            this.requestedEditMode = false;
+            this.toast.success(this.intl.t('registries.registration_metadata.edit_provider_metadata.success'));
+        }
+    });
+
+    @tracked currentModerator?: ModeratorModel;
```

Comment:
We may not need to fetch the currentModerator if we can check the provider and check `currentUserCanReview`. As it is right now, we don't pass in `@userCanEdit` when we invoke this manager either